rlp/main.cpp

```diff
@@ -256,14 +256,24 @@ int main(int argc, char** argv)
         prefs.escapeAll = true;
 
     bytes in;
-    if (inputFile == "--")
-        for (int i = cin.get(); i != -1; i = cin.get())
-            in.push_back((byte)i);
-    else if (boost::filesystem::is_regular_file(inputFile))
-        in = contents(inputFile);
-    else
-        in = asBytes(inputFile);
-
+    try
+    {
+        if (inputFile == "--")
+            for (int i = cin.get(); i != -1; i = cin.get())
+                in.push_back((byte)i);
+        else if (boost::filesystem::is_regular_file(inputFile))
+            in = contents(inputFile);
+    }
+    catch(const boost::filesystem::filesystem_error& _e)
+    {
+        //temporary workaround by comparing exception message with inputFile
+        string error_e = _e.what();
+        string long_file_name = "boost::filesystem::status: File name too long: \"";
+        long_file_name += inputFile + "\"";
+        if(error_e == long_file_name)
```

Comment:
You can probably assume that the input should be treated as bytecode if you get **any** error from `is_regular_file`.  Probably don't need to check the error message or type.

Reply:
And checking for error types by comparing against hard coded strings is bad form because it will break if the error message is changed upstream in Boost.

Reply:
Yeah I totally agree with you regarding the hard coded strings and that change in upstream boost will break code.
But,
I thought that in here
https://github.com/ethereum/aleth/blob/b09e9ad00b1f34259c0702ca521cd2e5585d526c/rlp/main.cpp#L259-L265
contents() internally call
https://github.com/ethereum/aleth/blob/ad7204c9ae7a13359520b7c21ffd8fc7484cf589/libdevcore/CommonIO.cpp#L86-L105
contentsGeneric() and here it's performing boost::filesystem operations so boost::filesystem::filesystem_error may occur and then catching every error and handling them as filename_too_long will create another bug.

Please correct me if I am wrong.
Also I have some ideas, but I need some time to select the best out of them.